packages/astro/src/vite-plugin-markdown/images.ts

```diff
@@ -4,7 +4,10 @@ export function getMarkdownCodeForImages(imagePaths: MarkdownImagePath[], html:
 	return `
 			import { getImage } from "astro:assets";
 			${imagePaths
-				.map((entry) => `import Astro__${entry.safeName} from ${JSON.stringify(entry.raw)};`)
+				.map((entry) => {
```

Comment:
Hmm, does this work? Doesn't this here https://github.com/withastro/astro/blob/eed0e8757c35dde549707e71c45862438a043fb0/packages/markdown/remark/src/remark-collect-images.ts#L30 prevents things from working anyway?

Reply:
When `![](img.jpg)` is used in post.md where the file path is `content/blog/post.md` `content/blog/img.jpg` `shouldOptimizeImage()` from `remark-collect-images.ts` returns true 

The failure at least from my investigation comes from the import 

I tested this by copying one of the images from public into the content folder in the astro/examples/blog project